Providers/Profile.swift

```diff
@@ -782,8 +806,59 @@ public class BrowserProfile: Profile {
             }
         }
 
+        @objc func onStartSyncing(notification: NSNotification) {
+            syncDisplayState = .InProgress
+        }
+
         @objc func onFinishSyncing(notification: NSNotification) {
-            self.lastSyncFinishTime = NSDate.now()
+            if let syncState = syncDisplayState where syncState == .Good {
+                self.lastSyncFinishTime = NSDate.now()
+            }
+        }
+
+        private func displayStateForSyncResult(result: SyncResult) -> SyncDisplayState {
+            guard result.value.isSuccess,
```

Comment:
In general we should avoid using `.value` on the Deferred objects since if the `Deferred` isn't filled, this will force it to synchronously be filled if unfilled. Although it's already filled when reaching this statement, if we ever refactor the code around we could be causing performance issues by blocking on this.